File: android/app/src/main/java/com/mattermost/share/ShareActivity.java
package com.mattermost.share;

import android.os.Bundle;

import com.facebook.react.ReactActivity;
import com.mattermost.rnbeta.MainApplication;

public class ShareActivity extends ReactActivity {
    @Override
    protected String getMainComponentName() {
        return "MattermostShare";
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MainApplication app = (MainApplication) this.getApplication();
        app.sharedExtensionIsOpened = true;
    }
}

File: android/app/src/main/java/com/mattermost/share/ShareActivity.kt
package com.mattermost.share

import android.os.Bundle
import com.facebook.react.ReactActivity
import com.mattermost.rnbeta.MainApplication

class ShareActivity : ReactActivity() {
    override fun getMainComponentName(): String? = "MattermostShare"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val app: MainApplication = this.application as MainApplication
        app.sharedExtensionIsOpened = true
    }
}
